hub/gads-ui/src/components/Admin/Workspaces/WorkspacesAdministration.js

```diff
@@ -0,0 +1,290 @@
+                            page={page}
+                            onPageChange={handleChangePage}
+                            onRowsPerPageChange={handleChangeRowsPerPage}
+                        />
+                    </TableContainer>
+                )}
+
+                <Modal open={openModal} onClose={handleCloseModal}>
+                    <Box sx={{
+                        padding: 4,
+                        backgroundColor: 'white',
+                        borderRadius: 2,
+                        boxShadow: 3,
+                        maxWidth: 400,
+                        margin: 'auto',
+                        mt: 5
+                    }}>
+                        <h2>{currentWorkspace ? 'Edit Workspace' : 'Add Workspace'}</h2>
+                        <Stack spacing={2}>
+                            <TextField
+                                label='Workspace Name'
+                                value={currentWorkspace ? currentWorkspace.name : newWorkspace.name}
+                                onChange={(e) => {
+                                    if (currentWorkspace) {
+                                        setCurrentWorkspace({ ...currentWorkspace, name: e.target.value });
+                                    } else {
+                                        setNewWorkspace({ ...newWorkspace, name: e.target.value });
+                                    }
+                                }}
+                                required
+                            />
+                            <TextField
+                                label='Description'
+                                value={currentWorkspace ? currentWorkspace.description : newWorkspace.description}
+                                onChange={(e) => {
+                                    if (currentWorkspace) {
+                                        setCurrentWorkspace({ ...currentWorkspace, description: e.target.value });
+                                    } else {
+                                        setNewWorkspace({ ...newWorkspace, description: e.target.value });
+                                    }
+                                }}
+                                required
+                            />
+                            <Stack direction='row' spacing={1}>
+                                <Button variant='contained' onClick={currentWorkspace ? handleUpdateWorkspace : handleCreateWorkspace}>
+                                    {currentWorkspace ? 'Apply' : 'Create'}
+                                </Button>
+                                <Button variant='outlined' onClick={handleCloseModal}>
+                                    Cancel
+                                </Button>
+                            </Stack>
+                        </Stack>
+                    </Box>
+                </Modal>
+
+                <Modal open={openDeleteModal} onClose={handleCloseDeleteModal}>
```

Comment:
We already have a reusable dialog for such actions applied across the whole app - you can check how it is used [here](https://github.com/shamanec/GADS/blob/c91ff7c85346b3bbb56da223bdaeab0096bf0308/hub/gads-ui/src/components/Admin/Users/UsersAdministration.js#L220)

Reply:
Done in [commit](https://github.com/shamanec/GADS/pull/174/commits/f442d635999eae72c3da43fc681d2be449b18be9)